tests/infra/ccf.py

```diff
@@ -318,9 +316,13 @@ def create_and_add_pending_node(self, lib_name, host, args, target_node=None):
             LOG.error(f"New pending node {new_node.node_id} failed to join the network")
             errors = new_node.stop()
             if errors:
+                # Add error lines from file as custom field on this error
+                err.joining_errors = []
                 for error in errors:
                     if "An error occurred while joining the network" in error:
-                        err.message = f"TimeoutError: {error.split('|', 1)[1]}"
```

Comment:
Perhaps unnecessary, but hopefully now clearer what's happening here. `err.message` looks like a standard field that we're overwriting, but its not, so we give it a more clearly custom name. And we save all the joining errors, rather than just the last.

Reply:
As suggested at the call site, let’s throw a specific exception here, it will be clearer.